src/engine/Server.cpp

```diff
@@ -626,20 +645,14 @@ boost::asio::awaitable<void> Server::processQuery(
     QueryExecutionContext qec(index_, &cache_, allocator_,
                               sortPerformanceEstimator_,
                               std::ref(messageSender), pinSubtrees, pinResult);
+    auto [cancellationHandle, cancelCancellationHandle] =
```

Comment:
If you only have the name, then it should be clear from the name or from the comment that the second one is actually a `Cleanup` that cancels the handle *on destruction*. Otherwise this code looks like it makes no sense, although it does:)